### PR TITLE
[igxPivot] Add empty dimension and values template.

### LINKED ISSUE #10977: [Pivot] Implement empty template for empty pivot configuration
Expose and render empty template when no values and dimensions are declared.

Also there should be no errors when the pivot configuration is empty

### PR DESCRIPTION
**Pull request by MayaKirova**
Closes #10977  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by mddragnev** [comment]
@MayaKirova 
When initially starting the dev sample an error is thrown with `ExpressionChangedAfterItHasBeenCheckedError` exception.
image


Also, when you open the grid and remove one by one every column and row dimension it makes them disabled but not entirely removing them from the configuration. In this case the message "Pivot grid has no dimensions and values defined" should not show.
Steps to reproduce:
1. Open pivot dev sample
2. Remove all dimensions from the chips

**Review by ChronosSF — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-filtering.service.ts [hunk 1]
```diff
@@ -21,8 +21,9 @@ export class IgxPivotFilteringService extends IgxFilteringService {
         super.clear_filter(fieldName);
         const grid = this.grid;
         const config = (grid as IgxPivotGridComponent).pivotConfiguration;
-        const allDimensions = PivotUtil.flatten(config.rows.concat(config.columns).concat(config.filters).filter(x => x !== null && x !== undefined));
-        const dim = allDimensions.find(x => x.memberName === fieldName || x.member === fieldName);
+        const allDimensions = (config.rows || []).concat((config.columns || [])).concat(config.filters || []).filter(x => x !== null && x !== undefined);
```
Comment (mddragnev) [inline projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-filtering.service.ts:24]: You can use `allDimension` getter from the grid component

File: projects/igniteui-angular/src/lib/data-operations/pivot-strategy.ts [hunk 2]
```diff
@@ -227,7 +227,7 @@ export class DefaultPivotSortingStrategy extends DefaultSortingStrategy {
         grid?: PivotGridType) {
         const key = fieldName;
         const config = grid.pivotConfiguration;
-        const allDimensions = config.rows.concat(config.columns).concat(config.filters).filter(x => x !== null && x !== undefined);
+        const allDimensions = (config.rows || []).concat((config.columns || [])).concat(config.filters || []).filter(x => x !== null && x !== undefined);
```
Comment (mddragnev) [inline projects/igniteui-angular/src/lib/data-operations/pivot-strategy.ts:230]: You can use `allDimension` getter from the grid component

File: projects/igniteui-angular/src/lib/data-operations/pivot-strategy.ts [hunk 3]
```diff
@@ -204,7 +204,7 @@ export class DimensionValuesFilteringStrategy extends FilteringStrategy {
     protected getFieldValue(rec: any, fieldName: string, isDate: boolean = false, isTime: boolean = false,
         grid?: PivotGridType): any {
         const config = grid.pivotConfiguration;
-        const allDimensions = config.rows.concat(config.columns).concat(config.filters).filter(x => x !== null && x !== undefined);
+        const allDimensions = (config.rows || []).concat((config.columns || [])).concat(config.filters || []).filter(x => x !== null && x !== undefined);
```
Comment (mddragnev) [inline projects/igniteui-angular/src/lib/data-operations/pivot-strategy.ts:207]: You can use `allDimension` getter from the grid component

**Review by ChronosSF — APPROVED** [review]

**Pull request merged**